proposals/0133-syscall-get-epoch-stake.md

```diff
@@ -67,32 +76,44 @@ uint64_t sol_get_epoch_stake(/* r1 */ void const * vote_addr);
 
 ### Control Flow
 
-The syscall aborts the virtual machine if not all bytes in VM memory range
-`[vote_addr, vote_addr + 32)` are readable.
+If `var_addr` is _not_ a null pointer:
 
-Otherwise, the syscall returns a `u64` integer representing the total active
-stake delegated to the vote account at the provided address.
+- The syscall aborts the virtual machine if not all bytes in VM memory range
+  `[vote_addr, vote_addr + 32)` are readable.
+- Otherwise, the syscall returns a `u64` integer representing the total active
+  stake delegated to the vote account at the provided address.
+  If the provided vote address corresponds to an account that is not a vote
+  account or does not exist, the syscall will return `0` for active stake.
 
-If the provided vote address corresponds to an account that is not a vote
-account or does not exist, the syscall will return `0` for active stake.
+If `var_addr` is a null pointer, the syscall returns a `u64` integer
+representing the total active stake on the cluster for the current epoch.
 
 ### Compute Unit Usage
 
-The syscall will always attempt to consume the same amount of CUs regardless of
-control flow.
+The syscall will always attempt to consume the same amount of CUs for both
+possible `var_addr` pointers, regardless of control flow.
+
+If `var_addr` is _not_ a null pointer:
+
+```
+syscall_base + floor(PUBKEY_BYTES/cpi_bytes_per_unit) + mem_op_base
+```
+
+If `var_addr` is a null pointer:
 
 ```
-syscall_base + floor(32/cpi_bytes_per_unit) + mem_op_base
+syscall_base + mem_op_base
```

Comment:
I think this makes sense. I'll add this change.

@CriesofCarrots just to touch on our discussion about compute usage, where we were ideating on charging more compute to perform the summation, I think just `syscall_base` does make sense here. Caching the total epoch stake or otherwise structuring the runtime component to serve this syscall in a performant fashion seems like it should be an implementation detail.

Furthermore, we already do cache this information, so this call is trivial for Agave.

https://github.com/anza-xyz/agave/blob/9dd01f55182ff75a9c21bf89958ba5b37c8fe937/runtime/src/epoch_stakes.rs#L47-L49